web/packages/build/vite/config.ts

```diff
@@ -200,6 +196,20 @@ export function createViteConfig(
   });
 }
 
+function resolveAllowedHost(target: string) {
+  if (process.env.VITE_HOST) {
+    return [process.env.VITE_HOST];
+  }
+
+  if (target !== DEFAULT_PROXY_TARGET) {
+    const { hostname } = new URL(`http://${target}`);
+
+    return [hostname];
+  }
+
+  return [];
+}
```

Comment:
Could be something like this to allow `target` and `VITE_HOST` at the same time because why not.

```typescript
function resolveAllowedHost(target: string) {
  const allowedHosts = [];
  if (process.env.VITE_HOST) {
    allowedHosts.push(process.env.VITE_HOST);
  }

  if (target !== DEFAULT_PROXY_TARGET) {
    const { hostname } = new URL(`http://${target}`);

    allowedHosts.push(hostname);
  }

  return allowedHosts;
}
```

I guess it all comes down to which domains you have in the cert that Vite uses? But I noticed that the docs say "When using HTTPS, this check is skipped" and that is true, but the thing that falls apart without `allowedHosts` is websockets (and thus HMR).

Reply:
Yeah I think if your cert accepts `localhost` it'll still work without setting `allowedHosts` - but first Vite will try the hostname you're using, so there will be one error in the console always, but HMR would still work

I think your suggestion is good, I'll add it